bugsnag-android-core/src/main/java/com/bugsnag/android/RemoteConfig.kt

```diff
@@ -0,0 +1,136 @@
+import com.bugsnag.android.RemoteConfig.Companion.DISCARD_RULE_ALL_HANDLED
+import com.bugsnag.android.RemoteConfig.Companion.MATCH_TYPE
+import com.bugsnag.android.internal.DateUtils
+import java.util.Date
+
+internal class RemoteConfig(
+    val configurationTag: String,
+    val configurationExpiry: Date,
+    val discardRules: List<DiscardRule>
+) : JsonStream.Streamable {
+
+    override fun toStream(stream: JsonStream) {
+        stream.beginObject()
+        stream.name(KEY_CONFIGURATION_TAG).value(configurationTag)
+        stream.name(KEY_CONFIG_EXPIRY).value(DateUtils.toIso8601(configurationExpiry))
+        stream.name(KEY_DISCARD_RULES).beginArray()
+        for (rule in discardRules) {
+            stream.beginObject()
+            when (rule) {
+                is DiscardRule.All -> stream.value(rule)
+                is DiscardRule.AllHandled -> stream.value(rule)
+            }
+            stream.endObject()
+        }
+        stream.endArray()
+        stream.endObject()
+    }
+
+    internal companion object : JsonReadable<RemoteConfig?> {
+        internal const val KEY_CONFIGURATION_TAG = "configurationTag"
+        internal const val KEY_CONFIG_EXPIRY = "configurationExpiry"
+        internal const val KEY_DISCARD_RULES = "discardRules"
+        internal const val DISCARD_RULE_ALL = "ALL"
+        internal const val DISCARD_RULE_ALL_HANDLED = "ALL_HANDLED"
+        internal const val MATCH_TYPE = "match_type"
+
+        override fun fromReader(reader: JsonReader): RemoteConfig? {
+            var configurationTag: String? = null
+            var configurationExpiry: Date? = null
+            var discardRules: List<DiscardRule> = emptyList()
+            with(reader) {
+                beginObject()
+                while (hasNext()) {
+                    val key = nextName()
+                    when (key) {
+                        KEY_CONFIGURATION_TAG -> configurationTag = nextString()
+                        KEY_CONFIG_EXPIRY ->
+                            configurationExpiry = DateUtils.fromIso8601(nextString())
+
+                        KEY_DISCARD_RULES -> discardRules = parseDiscardRules(reader)
+                    }
+                }
+                endObject()
+            }
+            return if (configurationTag != null && configurationExpiry != null) {
+                RemoteConfig(
+                    configurationTag = configurationTag!!,
+                    configurationExpiry = configurationExpiry!!,
+                    discardRules
+                )
+            } else {
+                null
+            }
+        }
+
+        private fun parseDiscardRules(reader: JsonReader): List<DiscardRule> {
+            val rules = ArrayList<DiscardRule>()
+            var rule: DiscardRule?
+            with(reader) {
+                beginObject()
+                beginArray()
+                while (hasNext()) {
+                    rule = DiscardRule.fromReader(reader)
```

Comment:
the declaration of `rule` should be with it assignment:
```suggestion
            with(reader) {
                beginObject()
                beginArray()
                while (hasNext()) {
                    val rule = DiscardRule.fromReader(reader)
```